src/i2c.rs

```diff
@@ -436,12 +436,17 @@ where
         last_ret
     }
 
-    fn write_without_stop(&mut self, addr: u8, bytes: &[u8]) -> NbResult<(), Error> {
-        self.send_start_and_wait()?;
+    fn send_addr_and_wait(&mut self, addr: u8, read: bool) -> NbResult<(), Error> {
         self.nb.i2c.sr1.read();
```

Comment:
Not directly related to the PR, but I'm now becomming unsure about what this is doing here. My best guess is that it tries to clear the status registers, but clearly that's not done since the first commit in the PR is needed...

Reply:
hmm; looking at the datasheet i'm not entirely sure if it's needed here either; Fwiw the relevant parts are:
```
Then the master waits for a read of the SR1 register followed by a write in the DR register
with the Slave address (see Figure 273 and Figure 274 Transfer sequencing EV5).
```
with EV5 being `EV5: SB=1, cleared by reading SR1 register followed by writing DR register with Address.`

But ofcourse the `send_start_and_wait` is already reading SR1 while polling for  the flag; So it's probably not needed here.

Overall it seems some bits of sr1 are persistents (the error bits in particular) while others gets cleared on read

Reply:
Right. I think the best course of action here is to break this into its own issue and maybe fix it later since doing the wrong thing could break things subtly.